File: 30-days-of-code/day-28-regex,-patterns,-and-intro-to-databases/solution.py
#!/bin/python3

import sys
import re

res = []
def print_list(arr):
    for s in arr:
        print(s)

def is_match(email):
    p = re.compile('.*@gmail.com$')
    if re.search(p, email):
        return True
    return False

N = int(input().strip())
for a0 in range(N):
    firstName,emailID = input().strip().split(' ')
    firstName,emailID = [str(firstName),str(emailID)]

    if is_match(emailID):
        res.append(firstName)

res.sort()
print_list(res)

File: 30-days-of-code/day-28-regex,-patterns,-and-intro-to-databases/solution.js
process.stdin.resume();
process.stdin.setEncoding('ascii');

var input_stdin = "";
var input_stdin_array = "";
var input_currentline = 0;

process.stdin.on('data', function (data) {
    input_stdin += data;
});

process.stdin.on('end', function () {
    input_stdin_array = input_stdin.split("\n");
    main();
});

function readLine() {
    return input_stdin_array[input_currentline++];
}

/////////////// ignore above this line ////////////////////

function main() {
    var N = parseInt(readLine());
    let res = [];

    for(var a0 = 0; a0 < N; a0++){
        var firstName_temp = readLine().split(' ');
        var firstName = firstName_temp[0];
        var emailID = firstName_temp[1];

        if (isMatch(emailID)) res.push(firstName);
    }

    res.sort();
    printArray(res);
}

function printArray(arr) {
    for (let s of arr) {
        console.log(s);
    }
}

function isMatch(email) {
    const regEx = /.*@gmail.com$/;

    if (regEx.test(email)) return true;
    return false;
}
